sdk/identity/azure-identity/src/managed_identity_source.cpp

```diff
@@ -212,60 +242,61 @@ Azure::Core::Credentials::AccessToken AzureArcManagedIdentitySource::GetToken(
     {
       auto& httpRequest = request->HttpRequest;
       httpRequest.SetHeader("Metadata", "true");
+
+      if (!scopesStr.empty())
       {
-        auto const& scopes = tokenRequestContext.Scopes;
-        if (!scopes.empty())
-        {
-          httpRequest.GetUrl().AppendQueryParameter("resource", FormatScopes(scopes, true));
-        }
+        httpRequest.GetUrl().AppendQueryParameter("resource", scopesStr);
       }
     }
 
     return request;
   };
 
-  return TokenCredentialImpl::GetToken(
-      context,
-      createRequest,
-      [&](auto const statusCode, auto const& response) -> std::unique_ptr<TokenRequest> {
-        using Core::Credentials::AuthenticationException;
-        using Core::Http::HttpStatusCode;
-
-        if (statusCode != HttpStatusCode::Unauthorized)
-        {
-          return nullptr;
-        }
-
-        auto const& headers = response.GetHeaders();
-        auto authHeader = headers.find("WWW-Authenticate");
-        if (authHeader == headers.end())
-        {
-          throw AuthenticationException(
-              "Did not receive expected WWW-Authenticate header "
-              "in the response from Azure Arc Managed Identity Endpoint.");
-        }
-
-        constexpr auto ChallengeValueSeparator = '=';
-        auto const& challenge = authHeader->second;
-        auto eq = challenge.find(ChallengeValueSeparator);
-        if (eq == std::string::npos
-            || challenge.find(ChallengeValueSeparator, eq + 1) != std::string::npos)
-        {
-          throw AuthenticationException("The WWW-Authenticate header in the response from Azure "
-                                        "Arc Managed Identity Endpoint "
-                                        "did not match the expected format.");
-        }
-
-        auto request = createRequest();
-        std::ifstream secretFile(challenge.substr(eq + 1));
-        request->HttpRequest.SetHeader(
-            "Authorization",
-            "Basic "
-                + std::string(
-                    std::istreambuf_iterator<char>(secretFile), std::istreambuf_iterator<char>()));
-
-        return request;
-      });
+  return TokenCache::GetToken(std::string(), GetClientId(), GetAuthorityHost(), scopesStr, [&]() {
```

Comment:
Why an explicit constructor for `std::string` instead of ""?

Reply:
Just a micro-optimization - default constructor would be faster than for the string class to read `const char*` arg to find out it is an empty string, plus `""` is not getting allocated in `.text`.